torchrl/data/rlhf/utils.py

```diff
@@ -87,7 +155,13 @@ class RolloutFromModel:
     EOS_TOKEN_ID = 50256
 
     def __init__(
-        self, model, ref_model, reward_model, max_new_tokens=50, score_clip=10.0
+        self,
+        model,
+        ref_model,
+        reward_model,
+        kl_controller,
```

Comment:
this is missing from the doc.

I'm not so sure about this kl_controller that is passed to the module. I feel it should be handled separately. It's like passing the lr_scheduler to the optimizer, the reason we don't do that is that it mixes responsibilities between modules. It gives the impression that one module has multiple responsibilities but it is less clear than doing things explicitly in the main code.

Reply:
are you suggesting we go back to passing just the kl coefficient?

Reply:
I think the KL controller should be another class, but then we have 2 options:
the KL controller changes the KL coefficienbt of the other class (like the LR scheduler changes the LR of the optimizer or the target param updaters in torchrl update the target params of the loss) or we explicitely pass the kl coef.
I think the first option is more "pytorch"-style

Reply:
fixed

Reply:
Should we not remove it then?